### PR TITLE
Slot element with tabindex=0 that generates a box should be focusable

### PR DESCRIPTION
**Pull request by knowler**
In Chromium, the `<slot>` element that is generating a box (i.e. overriding the default `display: contents`) and has its `tabindex` attribute set to `0` in this structure cannot be focused by tabbing. In Safari and Firefox, this works fine.

```html
<div>
    <template shadowrootmode=open>
        <slot tabindex=0 style="display: inline-block;"></slot>
    </template>
    Content
</div>
```

I’m borrowing utilities from other tests in the same `/shadow-dom/focus/` directory as well loosing following some of the other tests. Also, I think I might not have the correct help/spec link here — I this was from the other tests I was following.

If this seems like expected behaviour I’ll follow up with a bug report to Chromium, otherwise, I’ll file bug reports for Safari and Firefox.

### CODE REVIEW

**Comment by knowler** [comment]
@keithamus (or anyone else): would you be able to help me fix this test?

File: shadow-dom/focus/focus-slot-box-generated-tabindex-0.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>HTML Test: focus - slot with tabindex=0 that generates a box should be focusable</title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/interaction.html#sequential-focus-navigation">
```
Comment (keithamus) [inline shadow-dom/focus/focus-slot-box-generated-tabindex-0.html:4]: You can try adding 

```suggestion
<meta name=timeout content=long>
<link rel="help" href="https://html.spec.whatwg.org/multipage/interaction.html#sequential-focus-navigation">
```

which gives the test extra time to run, although I'd be surprised if it was taking that long to run.

Comment (knowler) [reply shadow-dom/focus/focus-slot-box-generated-tabindex-0.html:4]: Ok, it actually seems like all of the tests like this are timing out for me locally, but on wpt.fyi they are running correctly and as expected for this test: Chrome fails while Safari and Firefox pass. I think I _might_ have the wrong help/spec link in the test (it was copied from another test), but I think this is ready for review now. Thanks for the help though!

**Review by keithamus — APPROVED** [review]
This LGTM :+1:

**Pull request merged**